pkg/sql/authorization.go

```diff
@@ -940,3 +943,107 @@ func insufficientPrivilegeError(
 		"user %s does not have %s privilege on %s %s",
 		user, kind, typeForError, object.GetName())
 }
+
+// WarmSyntheticPrivilegeCacheForVirtualTables will attempt to access
+// each virtual table, and, in doing so, will cache the privileges for those
+// virtual tables. This work is done in parallel. It aids in reducing the cold
+// start latency when accessing virtual tables which themselves list virtual
+// tables.
+func WarmSyntheticPrivilegeCacheForVirtualTables(
+	ctx context.Context,
+	stopper *stop.Stopper,
+	version clusterversion.Handle,
+	cache *cacheutil.Cache,
+	schemas catalog.VirtualSchemas,
+	ief descs.TxnManager,
+	db *kv.DB,
+) {
+	if !version.IsActive(ctx, clusterversion.V22_2SystemPrivilegesTable) {
+		return
+	}
+
+	start := timeutil.Now()
+	var err error
+	defer func() {
+		if err != nil {
+			log.Warningf(ctx, "failed to warm privileges for virtual tables: %v", err)
+		} else {
+			log.Infof(ctx, "warmed privileges for virtual tables in %v", timeutil.Since(start))
+		}
+	}()
+
+	var tableVersions []descpb.DescriptorVersion
+	vtablePathToPrivilegeDescs := make(map[string]*catpb.PrivilegeDescriptor)
+	query := fmt.Sprintf(
+		`SELECT path, username, privileges, grant_options FROM system.%s WHERE path LIKE '/%s/%%'`,
```

Comment:
I just realized that we don't have an index on `path`. That seems bad in general -- it means we always do a full table scan.

Reply:
filed https://github.com/cockroachdb/cockroach/issues/93525